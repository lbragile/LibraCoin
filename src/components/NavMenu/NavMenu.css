nav {
  background-color: #eee;
}

.nav-link {
  font-size: larger;
  text-align: right;
  color: black;
  font-weight: bolder;
}

.nav-link:hover,
.nav-link.active {
  font-weight: bolder;
}

.nav-link:hover {
  border-bottom: #8ac0ff 1px solid;
}

@media only screen and (max-width: 1200px) {
  .nav-link:hover {
    border-bottom: none;
  }
}
