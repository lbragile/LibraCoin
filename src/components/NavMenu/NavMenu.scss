$navbar_bg: #eee;
$nav_link_underscore: #8ac0ff;

nav {
  background-color: $navbar_bg;
}

.nav-link {
  font-size: larger;
  text-align: right;
  color: black;
  font-weight: bolder;

  &:hover,
  &.active {
    font-weight: bolder;
  }

  &:hover {
    border-bottom: $nav_link_underscore 1px solid;
  }
}

@media only screen and (max-width: 1200px) {
  .nav-link:hover {
    border-bottom: none;
  }
}
