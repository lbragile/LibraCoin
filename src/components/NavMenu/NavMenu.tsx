import React from "react";
import { Navbar, Nav } from "react-bootstrap";
import { StyledNav, StyledNavLink } from "../../styles/NavMenuStyles";

const NavLinkWrapper = ({ text }: { text: string }) => {
  return (
    <StyledNavLink href={"/LibraCoin/" + text} active={location.href.includes("/" + text)}>
      {text[0].toUpperCase() + text.slice(1)}
    </StyledNavLink>
  );
};

export default function NavMenu(): JSX.Element {
  return (
    <StyledNav expand="lg" className="mb-3">
      <Navbar.Brand aria-label="LibraCoin Logo" href="https://github.com/lbragile/LibraCoin">
        <img src="./assets/libracoin-logo-256.png" alt="LibraCoin Logo" width="64" height="64" />
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="ml-auto">
          <NavLinkWrapper text="wallet" />
          <NavLinkWrapper text="mine" />
          <NavLinkWrapper text="blockchain" />
        </Nav>
      </Navbar.Collapse>
    </StyledNav>
  );
}
