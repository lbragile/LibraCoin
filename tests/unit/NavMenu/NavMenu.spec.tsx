import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import NavMenu from "../../../src/components/NavMenu/NavMenu";

interface IActiveStateProps {
  route: string;
  wallet: boolean;
  mine: boolean;
  blockchain: boolean;
}

it("renders correctly", () => {
  const { asFragment } = render(<NavMenu />);

  const walletLink = screen.getByText("Wallet");
  const mineLink = screen.getByText("Mine");
  const blockChainLink = screen.getByText("Blockchain");
  const brandLink = screen.getByRole("link", { name: /LibraCoin Logo/i }) as HTMLAnchorElement;
  const brandImg = screen.getByAltText(/LibraCoin Logo/i);

  // links & images render properly
  [walletLink, mineLink, blockChainLink, brandLink, brandImg].forEach((link) => {
    expect(link).toBeInTheDocument();
  });

  // links have correct href
  expect(walletLink).toHaveAttribute("href", "/LibraCoin/wallet");
  expect(mineLink).toHaveAttribute("href", "/LibraCoin/mine");
  expect(blockChainLink).toHaveAttribute("href", "/LibraCoin/blockchain");
  expect(brandLink).toHaveAttribute("href", "https://github.com/lbragile/LibraCoin");

  expect(asFragment()).toMatchSnapshot();
});

describe("Check active state of nav links for different URLs", () => {
  let baseURL: string;

  beforeAll(() => {
    baseURL = location.href;
    // make location.href be able to change
    Object.defineProperty(window, "location", { value: { href: baseURL }, writable: true });
  });

  afterAll(() => {
    Object.defineProperty(window, "location", { value: { href: baseURL } });
  });

  it.each`
    route           | wallet   | mine     | blockchain
    ${"wallet"}     | ${true}  | ${false} | ${false}
    ${"mine"}       | ${false} | ${true}  | ${false}
    ${"blockchain"} | ${false} | ${false} | ${true}
  `("sets $route link to active, all other's inactive", ({ route, wallet, mine, blockchain }: IActiveStateProps) => {
    const base_class = "nav-link ";
    location.href = baseURL + route;

    render(<NavMenu />);

    expect(screen.getByText("Wallet")).toHaveClass(base_class + (wallet ? "active" : ""));
    expect(screen.getByText("Mine")).toHaveClass(base_class + (mine ? "active" : ""));
    expect(screen.getByText("Blockchain")).toHaveClass(base_class + (blockchain ? "active" : ""));
  });
});
