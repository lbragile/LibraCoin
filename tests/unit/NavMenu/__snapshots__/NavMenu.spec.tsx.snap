// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #eee;
}

.c1 {
  font-size: larger;
  color: #000;
  font-weight: bolder;
  text-align: right;
}

.c1:hover,
.c1.active {
  font-weight: bolder;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: #8ac0ff;
  text-decoration-color: #8ac0ff;
  text-underline-offset: 8px;
}

@media only screen and (max-width:600px) {
  .c1:hover {
    -webkit-text-decoration: none;
    text-decoration: none;
  }
}

<nav
    class="c0 mb-3 navbar navbar-expand-lg navbar-light"
  >
    <a
      aria-label="LibraCoin Logo"
      class="navbar-brand"
      href="https://github.com/lbragile/LibraCoin"
    >
      <img
        alt="LibraCoin Logo"
        height="64"
        src="undefined/images/lc-192.png"
        width="64"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="ml-auto navbar-nav"
      >
        <a
          class="c1 nav-link"
          data-rb-event-key="/LibraCoin/wallet"
          href="/LibraCoin/wallet"
        >
          Wallet
        </a>
        <a
          class="c1 nav-link"
          data-rb-event-key="/LibraCoin/mine"
          href="/LibraCoin/mine"
        >
          Mine
        </a>
        <a
          class="c1 nav-link"
          data-rb-event-key="/LibraCoin/blockchain"
          href="/LibraCoin/blockchain"
        >
          Blockchain
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
